### PR TITLE
Adjust repo config for VS Code formatting

### LINKED ISSUE #122050: Disable `dart format` on save for VSCode contributors
We should add a default workspace settings file for VSCode (one of the primary editors used by Flutter contributors) to ensure that it doesn't format on save by default while contributors are making changes to the Flutter repository. 

It's frustratingly common that someone makes a good change, saves it, and then realizes to their horror that they have reformatted the entire class file. They then have to go through and revert the formatting changes while retaining the semantic changes they've made, a time-consuming and somewhat error-prone process. Adding this file to override the default VSCode settings will remove one tripwire for new contributors to the project.

### PR DESCRIPTION
**Pull request by guidezpl**
If someone's user settings include the following, the current workspace level formatting settings don't apply since they aren't as specific.
```
"[dart]": {
  "editor.formatOnSave": true,
  "editor.formatOnType": true,
}
```

This PR handles this by turning off auto-formatting specifically for Dart files.

This PR also adds `"files.trimTrailingWhitespace": true` to simplify [contribution guide](https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment#set-up-your-environment).

Fixes https://github.com/flutter/flutter/issues/122050. This change is test-exempt.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by guidezpl** [comment]
@DanTup is my understanding of user vs. workspace settings + language-specific settings correct?

**Review by DanTup — APPROVED** [review]
Yes, you're right - I overlooked this in the last review 🙈

We should also add `editor.formatOnPaste`. That's what I meant to include in the comment at https://github.com/flutter/flutter/pull/122048#discussion_r1135735313 but apparently just included formatOnSave again.

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,8 +4,11 @@
     // VSCode formats files on save by default. Since Flutter source code is
     // hand-formatted, the default settings are changed to prevent inadvertent
     // reformatting of code.
-    "editor.formatOnSave": false,
-    "editor.formatOnType": false,
+    "[dart]": {
+		"editor.formatOnSave": false,
+		"editor.formatOnType": false,
+        "editor.formatOnPaste": false,
```
Comment (DanTup) [inline .vscode/settings.json:10]: nit: indenting is different here.. some tabs, some spaces

**Review by timsneath — APPROVED** [review]
LGTM!

**Comment by Hixie** [comment]
test-exempt: configuration change

**Comment by timsneath** [comment]
@guidezpl I also noticed this:
https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment#set-up-your-environment

which suggests the following config change:
```
    "files.trimTrailingWhitespace": true
```

If we can add that one also, I think we can delete the wiki VSCode configuration section, which will reduce additional configuration toil for contributors. Any chance you can add this to this PR?

**Comment by guidezpl** [comment]
> If we can add that one also, I think we can delete the wiki VSCode configuration section, which will reduce additional configuration toil for contributors. Any chance you can add this to this PR?

Nice, I'll add it and edit the wiki.

**Pull request merged**